compiler/daml-lf-verify/src/DA/Daml/LF/Verify/Context.hs

```diff
@@ -0,0 +1,138 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Contexts for DAML LF static verification
+module DA.Daml.LF.Verify.Context
+  ( Delta
+  , MonadDelta, devars, devals
+  , UpdateSet(..)
+  , UpdCreate(..), usCre, usArc, usCho
+  , UpdArchive(..)
+  , UpdChoice(..)
+  , runDelta
+  , emptyDelta
+  , introDelta, extVarDelta
+  , lookupDExprVar, lookupDVal, lookupDChoice
+  , concatDelta
+  , emptyUpdateSet
+  , concatUpdateSet
+  ) where
+
+import Control.Lens hiding (Context)
+import Control.Monad.Error.Class (MonadError (..))
+import Control.Monad.Reader
+import Data.HashMap.Strict (HashMap, union, empty)
+
+import DA.Daml.LF.Ast
+import DA.Daml.LF.TypeChecker.Error
+
+data UpdCreate = UpdCreate
+  { _creTemp  :: !(Qualified TypeConName)
+    -- ^ Qualified type constructor corresponding to the contract template.
+  , _creField :: ![(FieldName, Expr)]
+    -- ^ The fields to be verified, together with their value.
+  }
+data UpdArchive = UpdArchive
+  { _arcTemp  :: !(Qualified TypeConName)
+    -- ^ Qualified type constructor corresponding to the contract template.
+  , _arcField :: ![(FieldName, Expr)]
+    -- ^ The fields to be verified, together with their value.
+  }
+data UpdChoice = UpdChoice
+  { _choTemp  :: !(Qualified TypeConName)
+    -- ^ Qualified type constructor corresponding to the contract template.
+  , _choName  :: !ChoiceName
+    -- ^ The name of the choice.
+  }
+
+-- | The List of updates being performed
+data UpdateSet = UpdateSet
+  { _usCre :: ![UpdCreate]
+    -- ^ The list of create updates.
+  , _usArc :: ![UpdArchive]
+    -- ^ The list of archive updates.
+  , _usCho :: ![UpdChoice]
+    -- ^ The list of choice updates.
+  }
+
+makeLenses ''UpdateSet
+
+emptyUpdateSet :: UpdateSet
+emptyUpdateSet = UpdateSet [] [] []
+
+concatUpdateSet :: UpdateSet -> UpdateSet -> UpdateSet
+concatUpdateSet (UpdateSet cres1 arcs1 chos1) (UpdateSet cres2 arcs2 chos2) =
+  UpdateSet (cres1 ++ cres2) (arcs1 ++ arcs2) (chos1 ++ chos2)
+
+-- | The environment for the DAML-LF verifier
```

Comment:
Is there some deeper meaning behind calling this `Delta` apart from the fact that this is commonly used in type theory? If not, maybe just call it `Environment` or some variation thereof?